gef.py

```diff
@@ -9269,11 +9269,24 @@ def build_line(self, name: str, color: str, address_val: int, got_address: int)
         return line
 
     @only_if_gdb_running
-    def do_invoke(self, argv: List[str]) -> None:
+    @parse_arguments({"symbols": [""]}, {"--all": False})
+    def do_invoke(self, _: List[str], **kwargs: Any) -> None:
+        args : argparse.Namespace = kwargs["arguments"]
+        if args.all:
+            vmmap = gef.memory.maps
+            mapfiles = set(mapfile.path for mapfile in vmmap if
+                           pathlib.Path(mapfile.path).is_file() and
+                           mapfile.permission & Permission.EXECUTE)
+            for mapfile in mapfiles:
+                self.do_invoke_for(mapfile, args.symbols)
+        else:
+            self.do_invoke_for(gef.session.file, args.symbols)
+
+    def do_invoke_for(self, file: str, argv: List[str]) -> None:
```

Comment:
Let's name this something better. It might not even need to be a separate function, but you might want to still break some of the functionality into helper functions.

Reply:
Do you have any suggestions?

Reply:
`print_got_for`?